### PR TITLE
[Feature][HiveSink] Support for HIVE collection data types.

### LINKED ISSUE #5447: [Feature][HiveSink] Support for HIVE collection data types.
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support for HIVE collection data types: Array,Map,Struct

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by lightzhao**
## Purpose of this pull request
close #5447 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by lightzhao** [comment]
@EricJoy2048 @ic4y @Hisoka-X PTAL.

**Comment by liugddx** [comment]
Can you add some test cases for this feature?

**Comment by lightzhao** [comment]
### step 1:create hive table
~~~
CREATE TABLE dbtest01.tmp_t_hive_data_type_test_2_day (
  id INT,
  name STRING,
  hobbies ARRAY<STRING>,
  address MAP<STRING, STRING>,
  details STRUCT<age: INT, email: STRING>
)
PARTITIONED BY (dt STRING)
ROW FORMAT DELIMITED
FIELDS TERMINATED BY '#'
COLLECTION ITEMS TERMINATED BY ','
MAP KEYS TERMINATED BY ':';
~~~
### step 2:seatunnel config
~~~
env {
    execution.parallelism = 1
    job.mode = "STREAMING"
    job.name = "test_hive_data_type"
    execution.planner  = "blink"
    execution.checkpoint.interval = 180000
  }

  source {
   Kafka {
    result_table_name = "source_table"
    format  = "text"
    field_delimiter =  "#"
    schema = {
      fields{
        id = string
        name = string
        hobbies = "array<string>"
        address = "map<string,string>"
        details =  {
          age = int
          email = string
        }
      }
    }
    topic  = "test5"
    bootstrap.servers  = "broker1:9092,broker2:9092,broker:9092"
   }
  }

  transform {
   sql {
    sql  = "select   `id`  as id, `name`  as name, `hobbies`  as hobbies, `address`  as address, `details`  as details, CAST(DATE_FORMAT(
CAST(NOW() AS VARCHAR),'yyyyMMdd') as VARCHAR)  as dt from source_table  "
   }
  }

  sink {
   Console{}
   Hive {
    table_name  = "dbtest01.tmp_t_hive_data_type_test_2_day"
    metastore_uri  = "thrift://localhost:9083"
   }
  }
~~~
### step 3: kafka produce data
image


### step4: hive query
image

**Comment by lightzhao** [comment]
> Can you add some test cases for this feature?

done, Screenshot as above.

File: docs/en/connector-v2/sink/Hive.md [hunk 1]
```diff
@@ -25,21 +25,24 @@ By default, we use 2PC commit to ensure `exactly-once`
   - [x] parquet
   - [x] orc
   - [x] json
+  - [x] hive_text
```
Comment (yx91490) [inline docs/en/connector-v2/sink/Hive.md]: Hi lightzhao, Should we document what's the differences between text and hive_text?

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
When the format of the hive table is text, the separation between collection and map increases by  \u001 based on the number of nested layers. In general production practice, this separator is not customized by the user because there may be nested data. This change make no sense.

File: docs/en/connector-v2/sink/Hive.md (hunk 1 above)
Comment (lightzhao) [reply docs/en/connector-v2/sink/Hive.md]: hive_text is compatible with the text storage format of hive collection data type, array, map, struct.

**Comment by lightzhao** [comment]
> When the format of the hive table is text, the separation between collection and map increases by \u001 based on the number of nested layers. In general production practice, this separator is not customized by the user because there may be nested data. This change make no sense.

In fact, the delimiter symbol can be customized by the user.
For collection data types in Hive, the default delimiter symbols are as follows:
The array type uses commas (",") as the separator between elements;
The map type uses colons (":") to separate key-value pairs, and then commas (",") to separate different key-value pairs;
The struct type uses commas (",") to separate fields in the structure.
The current delimiter symbols cannot adapt to the collection storage format of hive, and the queries are also confusing.

**Comment by TyrantLucifer** [comment]
You can see the implementation in text format.This change is duplicated

**Comment by lightzhao** [comment]
> You can see the implementation in text format.This change is duplicated

The current implementation fixes the separator symbols between elements. Hive’s default collection type cannot adapt. You can try it with the above configuration.

**Comment by hailin0** [comment]
Please fix code conflict

**Comment by melin** [comment]
support  orc/parquet ?

**Comment by lightzhao** [comment]
> support orc/parquet ?

This is only for collection data types in text format.

File: docs/en/connector-v2/sink/Hive.md (hunk 1 above)
Comment (Hisoka-X) [reply docs/en/connector-v2/sink/Hive.md]: I think you should metion it on the doc.

Comment (lightzhao) [reply docs/en/connector-v2/sink/Hive.md]: ok

File: docs/en/connector-v2/sink/Hive.md [hunk 2]
```diff
@@ -25,21 +25,24 @@ By default, we use 2PC commit to ensure `exactly-once`
   - [x] parquet
   - [x] orc
   - [x] json
+  - [x] hive_text : support the text storage format of hive collection data type, array, map, struct.
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/Hive.md:28]: `hive_text` is not a file format, can not put it here. You can use example to tell uses how to use it.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java [hunk 3]
```diff
@@ -100,10 +102,6 @@ public BaseFileSinkConfig(@NonNull Config config) {
             timeFormat =
                     TimeUtils.Formatter.parse(config.getString(BaseSinkConfig.TIME_FORMAT.key()));
         }
-
-        if (config.hasPath(BaseSinkConfig.ENABLE_HEADER_WRITE.key())) {
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java:104]: Why remove this?

**Pull request closed without merging**